src/Core/Utils/Index.hpp

```diff
@@ -16,7 +16,8 @@ class RA_CORE_API Index
     /// Default constructor that allow implicit conversion from integer to Index
     constexpr Index( IntegerType i = s_invalid );
     constexpr Index( const Index& i );
-    constexpr Index& operator=( long int i );
+    template <typename Integer>
```

Comment:
Please add static_assert to check if the type is integral, and mention this in the documentation.
Doc: https://en.cppreference.com/w/cpp/types/is_integral